File: DSAlgorithms/Leetcode/c++/arrays_hash/205-isomorphic_strings.cpp
bool isIsomorphic(string s, string t) {
    std::unordered_map<char, char> st;
    std::unordered_map<char, char> ts;
    for( int i = 0; i < s.size(); i++)
    {
        char c1 = s[i];
        char c2 = t[i];
        if((st.count(c1) > 0 && st[c1] != c2)||(ts.count(c2) > 0 && ts[c2] != c1))
            return false;
        st[c1] = c2;
        ts[c2] = c1;
    }
    return true;
}

File: DSAlgorithms/Leetcode/ts/src/arrays_hash/205-isomorphic_strings.ts
const isIsomorphic = (s: string, t: string): boolean =>{
    const s_map = new Map<string, string>();
    const t_map = new Map<string, string>();    
    for(let i = 0; i < s.length; i++)
    {
        const c1: string = s[i];
        const c2: string= t[i];
        if((s_map.has(c1) && s_map.get(c1) !== c2) || (t_map.has(c2) && t_map.get(c2) !== c1))
            return false;
        s_map.set(c1, c2);
        t_map.set(c2, c1);
    }
    return true;
};
